src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -1228,7 +1228,7 @@ protected function callScope(callable $scope, $parameters = [])
         // query as their own isolated nested where statement and avoid issues.
         $originalWhereCount = count($query->wheres);
 
-        $result = call_user_func_array($scope, $parameters) ?: $this;
+        $result = $scope(...$parameters) ?: $this;
```

Comment:
Don't know if we can guarantee only numeric keys on it. So, `...array_values($parameters)` can mitigate that. What you think?

I've pushed some changes, by the way.